test/FunctionalTests/Client/CancellationTests.cs

```diff
@@ -432,4 +434,321 @@ async Task ServerStreamingCall(DataMessage request, IServerStreamWriter<DataMess
             () => HasLog(LogLevel.Information, "GrpcStatusError", "Call failed with gRPC error status. Status code: 'Cancelled', Message: 'Call canceled by the client.'."),
             "Missing client cancellation log.").DefaultTimeout();
     }
+
+    [Test]
+    public async Task Unary_CancellationAfterDelay_TokenMatchesSource()
+    {
+        var tcs = new TaskCompletionSource<object?>(TaskCreationOptions.RunContinuationsAsynchronously);
+        async Task<DataMessage> UnaryMethod(DataMessage request, ServerCallContext context)
+        {
+            await tcs.Task;
+            return new DataMessage();
+        }
+
+        SetExpectedErrorsFilter(writeContext =>
+        {
+            return true;
+        });
+
+        // Arrange
+        var method = Fixture.DynamicGrpc.AddUnaryMethod<DataMessage, DataMessage>(UnaryMethod);
+        var channel = CreateChannel(throwOperationCanceledOnCancellation: true);
+        var client = TestClientFactory.Create(channel, method);
+
+        // Act
+        var cts = new CancellationTokenSource(TimeSpan.FromSeconds(0.5));
```

Comment:
I hate this, can you cancel the token inside `UnaryMethod` instead?

Reply:
Much better idea. All the other tests do this.